share/completions/history.fish

```diff
@@ -1,10 +1,33 @@
-complete -c history -r -l prefix --description "Match items starting with prefix"
-complete -c history -r -l contains --description "Match items containing string"
-complete -c history -l search -s s --description "Prints commands from history matching query"
-complete -c history -l delete -s d --description "Deletes commands from history matching query"
-complete -c history -l clear --description "Clears history file"
-complete -c history -l merge -s m --description "Incorporate history changes from other sessions"
-complete -c history -l exact -s e --description "Match items in the history that are identicial"
+set -g __fish_history_all_commands search delete save merge clear
+
+function __fish_history_cmd_in_array
+        for i in (commandline -pco)
+                # -- is used to provide no options for contains
+                # (if $i is equal to --optname without -- will be error)
+                if contains -- $i $argv
+                        return 0
+                end
+        end
+
+        return 1
+end
+
+function __fish_history_no_subcommand
+        not __fish_history_cmd_in_array $__fish_history_all_commands
+end
+
+complete -c history -l prefix -s p --description "Match items beginning with the string"
```

Comment:
Ahh, yes. That gives the effect I wanted but I didn't see how to do it despite having used `-n` below.